src/services/api.ts

```diff
@@ -28,16 +29,28 @@ export class APIService {
   }
 
   login = async (address: string): Promise<IApiLogin> => {
-    let now;
+    let nonce;
     try {
-      const nowReq = await fetch('/api/time');
-      now = parseInt(await nowReq.text());
-      if (isNaN(now)) now = Date.now();
+      const nonceReq = await fetch('/api/time');
+      nonce = await nonceReq.text();
     } catch (e) {
-      now = Date.now();
+      nonce = generateNonce();
     }
 
-    const data = `Login to Coordinape ${Math.floor(now / 1000)}`;
+    const message = new SiweMessage({
+      // TODO: replace by configured domain
+      domain: 'domain.tld',
```

Comment:
I've been thinking about CSRF attacks - would be great if we can have this domain be checked against the request origin, and be rejected if it doesn't match - to prevent someone from creating a phishing site that prompts for  this sign in message and then steals your api key.

Reply:
Part of the reason for SIWE is to work with wallet implementers on checking for domain binding with respect to parsing SIWE messages. We're currently working with a few on this at the moment.